src/plugins/AutoRefresh.ts

```diff
@@ -59,8 +61,10 @@ class AutoRefresh extends GenericPlugin {
       this.timeInView += 1;
 
       // Determine whether the ad should refresh
-      if (!this.isRefreshing && this.timeInView >= refreshRateInSeconds) {
+      if (!this.isRefreshing && this.timeInView >= refreshRateInSeconds
+        && (refreshLimit === null || this.refreshCount < refreshLimit)) {
```

Comment:
Can you assign this to a variable for readability. 
We care A LOT about bundle size in this application but the code is really complicated as is.

Reply:
Will do

Reply:
I just put the new code in a variable, let me know if you want all of it in a variable. Or if you want 2 separate variables.